sklearn/model_selection/_split.py

```diff
@@ -739,6 +739,15 @@ class TimeSeriesSplit(_BaseKFold):
     max_train_size : int, optional
         Maximum size for a single training set.
 
+    test_size : int, optional
+        Number of samples in each test set. Defaults to
+        ``n_samples / (n_splits + 1)``.
+
+    gap_size : int, default=0
```

Comment:
should we just call this `gap`?

Reply:
Would also like to contribute. The point of walk forward in my opinion is that you predict a certain time into the future. The number of folds or set per time series is irrelevant, since the prediction must get less accurate the bigger the gap. Therefore a fixed gap of 0 or more is a better practice than the number of folds in my opinion. If you compare different time series or use many time series created by the same process as the dataset, you might get different gaps if you divide the data using folds (based on number of samples) as opposed to comparable and static time gaps.

I also wanted to propose another feature: 
using different similar time series created by the same process for training and testing - and selection from that dataset for training and testing based on date and if the individual time series is already past or still current. Also a sliding window functionality would be great to create splits from!

...but this here already seems complex. What can I do to help get this walk forward feature integrated in TimeSeriesSplit?